awscli/autoprompt/factory.py

```diff
@@ -191,26 +243,51 @@ def create_layout(self, on_input_buffer_text_changed=None,
                 self.create_input_buffer(on_input_buffer_text_changed)
             input_buffer_container = \
                 self.create_input_buffer_container(input_buffer)
-        if doc_window is None:
-            doc_buffer = self.create_doc_buffer()
-            doc_window = self.create_doc_window(doc_buffer)
         if search_field is None:
             search_field = SearchToolbar()
+        if doc_window is None:
+            doc_buffer = self.create_doc_buffer()
+            doc_window = self.create_doc_window(doc_buffer, search_field)
         if bottom_toolbar_container is None:
             bottom_toolbar_buffer = self.create_bottom_toolbar_buffer()
             bottom_toolbar_container = \
                 self.create_bottom_toolbar_container(bottom_toolbar_buffer)
+        help_doc_buffer = self.create_help_buffer(
+            self._help_text.full_doc_window_key_binding_text
+        )
+        help_doc_window = self.create_help_window(
+            help_doc_buffer, title='Doc panel help')
+        help_input_buffer = self.create_help_buffer(
+            self._help_text.full_input_buffer_key_binding_text
+        )
+        help_input_window = self.create_help_window(
+            help_input_buffer, title='Prompt panel help')
         return Layout(
-            HSplit(
-                [
-                    input_buffer_container,
-                    ConditionalContainer(HorizontalLine(), doc_section_visible),
-                    ConditionalContainer(doc_window, doc_section_visible),
-                    HorizontalLine(),
-                    search_field,
-                    bottom_toolbar_container
-                ]
-            )
+            HSplit([
+                VSplit([
+                    HSplit(
+                        [
+                            input_buffer_container,
+                            ConditionalContainer(HorizontalLine(),
+                                                 doc_section_visible),
+                            ConditionalContainer(doc_window,
+                                                 doc_section_visible),
+                        ]
+                    ),
+                    ConditionalContainer(
+                        help_doc_window,
+                        help_section_visible & (doc_window_has_focus
+                        | search_input_has_focus)),
+                    ConditionalContainer(
+                        help_input_window,
+                        help_section_visible & input_buffer_has_focus),
+                ]),
+                search_field,
+                ConditionalContainer(HorizontalLine(),
+                                     ~help_section_visible),
+                ConditionalContainer(bottom_toolbar_container,
+                                     ~help_section_visible),
+            ])
```

Comment:
It would be nice if we can reduce the complexity of this top-level view of the app back to what we had before. Specifically, there are too many containers from this view and too much nesting. I think to address it, we can just make more helper functions that return top-level containers and hide the nesting in these helper functions and still have a nice higher-level view of the setup like we had before.